File: Trainee/Android/Java/AndroidTest/app/src/main/java/com/example/androidtest/MainActivity.java
package com.example.androidtest;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import androidx.core.content.ContextCompat;
import androidx.core.content.res.ResourcesCompat;
import androidx.fragment.app.Fragment;

import android.annotation.SuppressLint;
import android.graphics.Color;
import android.graphics.Typeface;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.androidtest.homefragment.EventFragment;
import com.example.androidtest.homefragment.ProfileFragment;
import com.example.androidtest.homefragment.SettingFragment;

public class MainActivity extends AppCompatActivity {

    CardView cvProfile, cvSetting, cvEvent;
    TextView tvProfile, tvSetting,tvEvent;
    ImageView imgProfile, imgSetting, imgEvent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        cvProfile = findViewById(R.id.cvProfile);
        cvSetting = findViewById(R.id.cvSetting);
        cvEvent = findViewById(R.id.cvEvent);
        tvProfile = findViewById(R.id.tvProfile);
        tvSetting = findViewById(R.id.tvSetting);
        tvEvent = findViewById(R.id.tvEvent);
        imgProfile = findViewById(R.id.imgProfile);
        imgSetting = findViewById(R.id.imgSetting);
        imgEvent = findViewById(R.id.imgEvent);

        selectFragment(new EventFragment(), imgEvent, tvEvent, cvEvent);

        cvEvent.setOnClickListener(v-> selectFragment(new EventFragment(), imgEvent, tvEvent, cvEvent));

        cvProfile.setOnClickListener(v-> selectFragment(new ProfileFragment(), imgProfile, tvProfile, cvProfile));

        cvSetting.setOnClickListener(v-> selectFragment(new SettingFragment(), imgSetting, tvSetting, cvSetting));
    }

    @SuppressLint("ResourceAsColor")
    private void selectFragment(Fragment fragment, ImageView imgView, TextView textView, CardView cardView){
        if (fragment != null) {

            resetNavigationBar();

            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.frmMain, fragment)
                    .commit();

            // set image color of selected fragment icon
            imgView.setColorFilter(ContextCompat.getColor(getApplicationContext(), R.color.white), android.graphics.PorterDuff.Mode.SRC_IN);

            // set font style of selected fragment text
            Typeface typeface = ResourcesCompat.getFont(getApplicationContext(), R.font.roboto_bold);
            textView.setTypeface(typeface);
            textView.setTextColor(Color.WHITE);

            // set card background tint
            cardView.setCardBackgroundColor(R.color.purple_background);
        }
    }

    @SuppressLint("ResourceAsColor")
    private void resetNavigationBar(){
        // set image color of selected fragment icon
        imgEvent.setColorFilter(ContextCompat.getColor(getApplicationContext(), R.color.black), android.graphics.PorterDuff.Mode.SRC_IN);
        imgProfile.setColorFilter(ContextCompat.getColor(getApplicationContext(), R.color.black), android.graphics.PorterDuff.Mode.SRC_IN);
        imgSetting.setColorFilter(ContextCompat.getColor(getApplicationContext(), R.color.black), android.graphics.PorterDuff.Mode.SRC_IN);

        // set font style of selected fragment text
        Typeface typeface = ResourcesCompat.getFont(getApplicationContext(), R.font.roboto_regular);
        tvEvent.setTypeface(typeface);
        tvEvent.setTextColor(Color.BLACK);

        tvProfile.setTypeface(typeface);
        tvProfile.setTextColor(Color.BLACK);

        tvSetting.setTypeface(typeface);
        tvSetting.setTextColor(Color.BLACK);

        // set card background tint
        cvEvent.setCardBackgroundColor(Color.TRANSPARENT);
        cvProfile.setCardBackgroundColor(Color.TRANSPARENT);
        cvSetting.setCardBackgroundColor(Color.TRANSPARENT);
    }
}

File: Trainee/Android/Java/Test2/app/src/main/java/com/example/test2/MainActivity.kt
package com.example.test2

import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageView
import androidx.cardview.widget.CardView
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import com.example.test2.bottomnav.HomeFragment
import com.example.test2.bottomnav.LogOutFragment
import com.example.test2.bottomnav.SettingFragment

class MainActivity : AppCompatActivity() {

    private lateinit var cvHome: CardView
    private lateinit var cvSetting: CardView
    private lateinit var cvLogout: CardView
    private lateinit var cvIndHome: CardView
    private lateinit var cvIndSetting: CardView
    private lateinit var cvIndLogout: CardView
    private lateinit var imgHome: ImageView
    private lateinit var imgSetting: ImageView
    private lateinit var imgLogout: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        cvHome = findViewById(R.id.cvHome)
        cvIndHome = findViewById(R.id.cvIndHome)
        imgHome = findViewById(R.id.imgHome)

        cvSetting = findViewById(R.id.cvSetting)
        cvIndSetting = findViewById(R.id.cvIndSetting)
        imgSetting = findViewById(R.id.imgSetting)

        cvLogout = findViewById(R.id.cvLogout)
        cvIndLogout = findViewById(R.id.cvIndLogout)
        imgLogout = findViewById(R.id.imgLogout)

        selectFragment(HomeFragment(), imgHome, cvIndHome)

        cvHome.setOnClickListener { selectFragment(HomeFragment(), imgHome, cvIndHome) }

        cvLogout.setOnClickListener { selectFragment(LogOutFragment(), imgLogout, cvIndLogout) }

        cvSetting.setOnClickListener { selectFragment(SettingFragment(), imgSetting, cvIndSetting) }
    }

    private fun selectFragment(fragment: Fragment, imgView: ImageView, cardIndView: CardView) {
        resetNavigationBar()

        supportFragmentManager.beginTransaction()
            .replace(R.id.frmMain, fragment)
            .commit()

        imgView.setColorFilter(ContextCompat.getColor(applicationContext, R.color.orange_light), android.graphics.PorterDuff.Mode.SRC_IN)

        cardIndView.setCardBackgroundColor(ContextCompat.getColor(applicationContext, R.color.orange_light))
    }

    private fun resetNavigationBar() {
        imgHome.setColorFilter(ContextCompat.getColor(applicationContext, R.color.black), android.graphics.PorterDuff.Mode.SRC_IN)
        imgLogout.setColorFilter(ContextCompat.getColor(applicationContext, R.color.black), android.graphics.PorterDuff.Mode.SRC_IN)
        imgSetting.setColorFilter(ContextCompat.getColor(applicationContext, R.color.black), android.graphics.PorterDuff.Mode.SRC_IN)

        cvHome.setCardBackgroundColor(Color.TRANSPARENT)
        cvLogout.setCardBackgroundColor(Color.TRANSPARENT)
        cvSetting.setCardBackgroundColor(Color.TRANSPARENT)

        cvIndHome.setCardBackgroundColor(Color.TRANSPARENT)
        cvIndLogout.setCardBackgroundColor(Color.TRANSPARENT)
        cvIndSetting.setCardBackgroundColor(Color.TRANSPARENT)
    }
}